nixos/modules/services/networking/znc.nix

```diff
@@ -379,7 +379,8 @@ in
         # If mutable, regenerate conf file every time.
         ${optionalString (!cfg.mutable) ''
           ${pkgs.coreutils}/bin/echo "znc is set to be system-managed. Now deleting old znc.conf file to be regenerated."
-          ${pkgs.coreutils}/bin/rm -f ${cfg.dataDir}/configs/znc.conf
+          ${pkgs.coreutils}/bin/echo "You can still recover you old config from '${cfg.dataDir}/configs/znc.conf.bak'."
+          ${pkgs.coreutils}/bin/mv ${cfg.dataDir}/configs/znc.conf{,.bak}
```

Comment:
I would only move it, if it is not a symlink to the nix store. Otherwise a second restart will overwrite you existing configuration. This also avoids moving if no mutable configuration has ever existed.

Reply:
It is never a symlink. If the file is not writeable, znc fails at startup. When it is not mutable, the file is overwritten at startup from the nix store file. Hence the issue.

Znc does not support immutable configuration files.

Reply:
We could copy it only if it differs from the reference store file.

Reply:
If it is overwritten on every startup anyway, we could also patch znc to stop complaining about read-only configuration.

Reply:
Patching ZNC seems the sane way to go, wit an error message when trying to change settings. I do not see how a patter would help, as probably ZNC overwrites the config on exit and on config change.

Reply:
Probably.

Reply:
Well, not going to happen soon... https://github.com/znc/znc/issues/975

I now think this is the safest way to go. Disabling immutability by default seems enough. This backup thing would have helped for me but is not that important if mutable is an opt-out.